examples/cross-contract-high-level/src/lib.rs

```diff
@@ -1,23 +1,24 @@
 use near_sdk::borsh::{self, BorshDeserialize, BorshSerialize};
 use near_sdk::{
-    env,
-    ext_contract,
-    json_types::U128,
-    log,
-    near_bindgen,
-    AccountId,
-    Promise,
-    PromiseOrValue,
+    env, ext_contract, json_types::U128, log, near_bindgen, AccountId, Gas, Promise, PromiseOrValue,
 };
 
+const TGAS: u64 = 1_000_000_000_000;
+
 #[near_bindgen]
 #[derive(Default, BorshDeserialize, BorshSerialize)]
 pub struct CrossContract {}
 
 // One can provide a name, e.g. `ext` to use for generated methods.
 #[ext_contract(ext)]
 pub trait ExtCrossContract {
-    fn merge_sort(&self, arr: Vec<u8>) -> PromiseOrValue<Vec<u8>>;
+    fn internal_merge_sort(&self, arr: Vec<u8>) -> PromiseOrValue<Vec<u8>>;
+    fn finalize_merge_sort(
```

Comment:
Can you explain why the change? is it so that you can call it with JSON params? Can you still not use borsh with runner? This example was already pretty unreadable, and this makes it even harder to follow imo

Reply:
The low level uses JSON and so with this change they could both have the same API and thus use the same tests.  We also discussed making a future PR that breaks up the cross contract example contracts into multiple ones that focus on just one aspect and not all in one.

Reply:
runner-js CAN work with borsh, but it's very ugly, and will be until we have an Application Contract Interface. 

Since this contract had been using borsh serialization, we wanted to keep that, but for the usability of both the tests and as a general example of the pattern, Willem and I thought it might be best to make it use JSON for external interfaces while saving gas by using Borsh for internal calls.

However, after completing this, we noticed that `cross-contract-low-level` does not use any Borsh, and I copy-pasted the same tests. So I know at this point that we could just remove Borsh from this example and have it work.

Honestly, I want to do a lot of rewriting of these examples:

* Make a "Factory Pattern" example, with a high-level and low-level implementation, that does the `status_message.wasm` deploy to subaccount. The same exact tests can run on each implementation, as a stronger guarantee of feature parity between the implementations (no borsh-on-one, not-on-the-other).
* Make a separate "Merge Sort" example, with a high-level and low-level implementation. Same thing with using the same tests for each implementation.
* Add good documentation for each of these, explaining what they demonstrate and why it's cool, rather than assuming that this is self-evident.

Given that we plan to give some love to all of these examples in the near future, we thought it was best to not change them too much right now. And the JSON-external, Borsh-internal pattern is a useful one, as is the `#[callback_unwrap] #[serializer(borsh)]`. We will probably want to keep these features in the reworked examples and just explain them better.

Reply:
All I'm saying is that it's very uncomfortable to change the example just to fit the test, especially since it just adds complexity.

I'm fine with this if we open an issue to make sure fixing and updating this doesn't slip under the rug though

Reply:
I removed all Borsh from cross-contract-high-level in dbc265a4f79232703d3225282ff7326f486f9eca and added two tracking issues for the examples reorganization, #589 & #590.